### PR TITLE
feat(readonlyvalue): new component readonlyvalue

### LINKED ISSUE #3124: [Component Proposal] Read-Only Value
### What package is this for?

- [x] React
- [ ] Angular

### Summary
A read-only value is information that is static and cannot be manipulated by users.

[Stacked variant](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/style#stacked-variant) →

image

[Default inline variant](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/style#default-inline-variant) →

image

[Small inline variant](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/style#small-inline-variant) →

image

------------------------------------------------------------------------------------------------------------------------

### Available resources
[PAL guidelines & Specs](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/usage)  →
[Link to sketch file](https://ibm.box.com/s/ntx935kkvp9redsee0wqyxldv3abiqmy) →

### PR DESCRIPTION
**Pull request by cgirani**
Closes #3124

**Summary**

- New component that displays information that is not editable in a similar structure as text inputs.  Though the information cannot be edited, a user can copy the displayed information.

**Stacked:**
image

**Inline:**
image

**Change List (commits, features, bugs, etc)**

- New component ReadOnlyValue

**Acceptance Test (how to verify the PR)**

- Validate that the components work and is presented as the [design specs](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/style) 

**Regression Test (how to make sure this PR doesn't break old functionality)**

- N/A

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/ReadOnlyValue/ReadOnlyValue.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# `ReadOnlyValue` component
+
+## Table of Contents
+
+- [Getting started](#getting-started)
+- [Variables](#variables)
```
Comment (davidicus) [inline packages/react/src/components/ReadOnlyValue/ReadOnlyValue.mdx:6]: There is no corresponding section for this part of the docs

**Review by davidicus — APPROVED** [review]
Looks good. Just a few minor questions and comments.

File: packages/react/src/components/ReadOnlyValue/ReadOnlyValue.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const ReadOnlyValue = ({ type, label, value, className, testId }) => {
+  return (
+    <div
+      data-testid={testId}
+      className={classnames(className, `${iotPrefix}--read-only-value`, {
+        [`${iotPrefix}--read-only-value__stacked`]: type === 'stacked',
+      })}
+    >
+      <p data-testid={`${testId}--label`}>{label}</p>
+      <span data-testid={`${testId}--value`}>{value}</span>
```
[the remark this answers is not in the document]
Comment (cgirani) [reply packages/react/src/components/ReadOnlyValue/ReadOnlyValue.jsx]: changed to be like this:
image

File: packages/react/src/components/ReadOnlyValue/ReadOnlyValue.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  label: PropTypes.string,
+  value: PropTypes.string,
+  testId: PropTypes.string,
+};
+
+const defaultProps = {
+  type: 'stacked',
+  label: '',
+  value: '',
+  testId: 'testId',
```
Comment (bjornalm) [inline packages/react/src/components/ReadOnlyValue/ReadOnlyValue.jsx:20]: We should use something less generic like `read-only-value` as the default

File: packages/react/src/components/ReadOnlyValue/ReadOnlyValue.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- [External links](#external-links)
+  - [Source Code](#source-code)
+  - [Feedback](#feedback)
+
+## Getting Started
+
+The `ReadOnlyValue` is a component for information that is static and cannot be manipulated by users.
+
+```jsx
+import { ReadOnlyValue } from 'carbon-addons-iot-react';
```
Comment (bjornalm) [inline packages/react/src/components/ReadOnlyValue/ReadOnlyValue.mdx:16]: I would add a small code example of how it is used, e.g:

```
    <ReadOnlyValue label="Label 01" value="input value 01" type="stacked" />
    <ReadOnlyValue label="Label 02" value="input value 02" type="stacked" />
```

File: packages/react/src/components/ReadOnlyValue/_read-only-value.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
+.#{$iot-prefix}--read-only-value {
+  display: block;
+  margin-bottom: $spacing-06;
+  label,
+  input {
+    display: block;
```
Comment (bjornalm) [inline packages/react/src/components/ReadOnlyValue/_read-only-value.scss:6]: There is an outline on focus-visible that we might want to remove:
image

```
&:focus-visible {
    outline: unset;
}
```

**Review by bjornalm — CHANGES_REQUESTED** [review]
Nice work, how does it handle really long values? Perhaps we can add knobs to the stories where the value and label can be modified?

**Comment by cgirani** [comment]
> Nice work, how does it handle really long values? Perhaps we can add knobs to the stories where the value and label can be modified?

It will change it to be a textarea and should render something like this
image

**Pull request merged**